src/overtone/sc/sclang.clj

```diff
@@ -0,0 +1,422 @@
+                             "check that SuperCollider is installed, https://supercollider.github.io/downloads")
+                        args)))
+
+      ;; Also, save a metadata file for the generated .scsyndef one so we
+      ;; can use it for caching. Also, people who don't have sclang available
+      ;; on their computers will be able to load the synthdef anyway as it's
+      ;; a normal file (assuming the ugens for external plugins are loaded ofc).
+      (spit metadata-file-path (binding [*print-length* nil
+                                         *print-level* nil]
+                                 (pr-str {:sc-clj sc-clj})))
+
+      (io/resource resource-path))))
+
+(comment
+
+  (def my-synth
+    (-> [:SynthDef {:name `event-2
+                    :args [[:freq 240] [:amp 0.5] [:pan 0.0]]}
+         [:vars :env]
+         [:= :env [:EnvGen.ar
+                   [:Env [0 1 1 0] [0.01 0.1 0.2]]
+                   {:doneAction 2}]]
+         [:Out.ar 0 [:Pan2.ar [:* [:Blip.ar :freq] :env :amp]
+                     :pan]]]
+        transpile
+        -synthdef-save!
+        ov.synth/synth-load))
+
+  (my-synth :freq 100)
+
+  ())
+
+(defn synth
+  "Defines a synth (SynthDef in SC) from a clojure data representation."
+  [sc-clj]
+  (-> sc-clj
+      transpile
+      -synthdef-save!
+      ov.synth/synth-load))
+
+(defn SynthDef
+  [opts & body]
+  (into [:SynthDef opts]
+        body))
+
+(comment
+
+  (def my-synth
+    (synth
+     (SynthDef
+      {:name 'event
+       :args [[:freq 120] [:amp 0.5] [:pan 0.0]]}
+      [:vars :env]
+      [:= :env [:EnvGen.ar
+                [:Env [0 1 1 0] [0.01 0.1 0.2]]
+                {:doneAction 2}]]
+      [:Out.ar 0 [:Pan2.ar [:* [:Blip.ar :freq] :env :amp]
+                  :pan]])))
+
+  (my-synth)
+
+  ())
+
+(defmacro defsynth
+  "Defines a synth (SynthDef in SC) from a clojure data representation.
+
+  This puts a metadata file and a .scsyndef file in a location derived from `s-name`
+  inside `resources-dir`.
+
+  `opts-map` can have the following optional keys
+    - `:args`, this is already derived from `params`, so you don't need to set it
+    - `:resources-dir`, default to \"resources\", it should be a resource folder
+
+  This will call your SuperColliders's `sclang` command at the first time while using `sc-clj`
+  for caching. As long you distribute the generated .scsyndef file in the right location and you don't
+  modify `sc-clj`, the final user won't need to have `sclang` or SuperCollider installed
+  on their machines.
+
+  Check `sclang-path` code for the default locations. You can also set the
+  `*sclang-user-path` atom to use an arbitrary `sclang` location.
+
+  --------------------
+  ;; Example.
+  ;; When you evaluate the form below, you should see new files in
+  ;; `resources/sc/synthdef`.
+  (sclang/defsynth my-synth
+    \"Some synth.\"
+    [freq 440, amp 0.5, pan 0.0]
```

Comment:
What do you think about asserting the params be keywords instead of symbols? No bindings are introduced. It might then also work out of the box with clj-kondo.

Reply:
I think using symbols is ok, and it's easier to visualize what are the args and what's the body, besides being as close as possible with the existing `defsynth. Let me know if this is fine \o

Reply:
Added a inline clj-kondo rule \o

Reply:
Yes looks good. Anyone who tries to refer to the symbols will get a compile-time error anyway.